MdeModulePkg/Include/Library/MemoryBucketHelperLib.h

```diff
@@ -0,0 +1,198 @@
+/** @file
+
+Library for defining PEI memory buckets. This keeps track of the
+different buckets and the data for the associated HOB.
+For internal use only.
+
+Copyright (C) Microsoft Corporation. All rights reserved.
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef __MEMORY_BUCKET_HELPER_LIB_H__
+#define __MEMORY_BUCKET_HELPER_LIB_H__
+
+#include <Uefi.h>
+#include <Base.h>
+#include <Guid/MemoryBucketInformation.h>
+
+/**
+  This function initialized the PEI memory buckets.  This should be called
+  when the first memory type being tracked in these buckets is allocated.
+
+  @param[in] StartingAddress   The starting address of the memory buckets.
+                               All of them are contiguous.
+**/
+VOID
+EFIAPI
+InitializeMemoryBuckets (
+  EFI_PHYSICAL_ADDRESS  StartingAddress

```

Comment:
Please add the `IN`, `OUT`, etc modifiers. This is especially useful for functions in header files.

Reply:
Updated all new functions to include IN/OUT.